subsys/nrf_security/Kconfig.legacy

```diff
@@ -21,8 +21,13 @@ config MBEDTLS_MEMORY_BUFFER_ALLOC_C
 	default y
 
 config MBEDTLS_THREADING_C
-	bool
+	bool "Threading support for Mbed TLS and PSA crypto"
 	default y if CC3XX_BACKEND || MBEDTLS_PSA_CRYPTO_C
+	help
+	  Threading support is used when PSA crypto is built locally or
+	  when legacy APIs implemented using CryptoCell is in use.
+	  Note: If this configuration is disabled then PSA crypto APIs
+	  aren't thread-safe. This configuration is not used when TF-M is enabled.
```

Comment:
Why not rather make it `depends on !BUILD_WITH_TFM`? Seems like it would make more sense.